spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java

```diff
@@ -188,6 +220,40 @@ public void onChangeEvent(ConfigurationChangeEvent event) {
             LOGGER.info("{} config changed, old value:{}, new value:{}", ConfigurationKeys.DISABLE_GLOBAL_TRANSACTION,
                 disable, event.getNewValue());
             disable = Boolean.parseBoolean(event.getNewValue().trim());
+        } else if (ConfigurationKeys.CLIENT_DEGRADE_CHECK.equals(event.getDataId())) {
+            degradeCheck = Boolean.parseBoolean(event.getNewValue());
+            if (!degradeCheck) {
+                degradeNum = 0;
+            }
+        }
+    }
+
+    /**
+     * auto upgrade service detection
+     */
+    private static void startSelfCheck() {
+        executor.scheduleAtFixedRate(() -> {
+            if (degradeCheck) {
+                try {
+                    String xid = TransactionManagerHolder.get().begin(null, null, "test", 60000);
+                    TransactionManagerHolder.get().commit(xid);
+                    onSelfCheck(true);
+                } catch (Exception e) {
+                    onSelfCheck(false);
+                }
+            }
+        }, degradeCheckPeriod, degradeCheckPeriod, TimeUnit.MILLISECONDS);
+    }
+
+    private static synchronized void onSelfCheck(boolean succeed) {
+        if (succeed) {
+            if (degradeNum > 0) {
+                degradeNum--;
+            }
+        } else {
+            if (degradeNum < degradeCheckAllowTimes) {
+                degradeNum++;
```

Comment:
degradeNum will never be greater than degradeCheckAllowTimes, how to trigger a degrade？

Reply:
> degradeNum will never be greater than degradeCheckAllowTimes, how to trigger a degrade？

PTAL